File: Python/decode-xored-array.py
# Time:  O(n)
# Space: O(1)

class Solution(object):
    def decode(self, encoded, first):
        """
        :type encoded: List[int]
        :type first: int
        :rtype: List[int]
        """
        result = [first]
        for x in encoded:
            result.append(result[-1]^x)
        return result

File: C++/decode-xored-array.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
public:
    vector<int> decode(vector<int>& encoded, int first) {
        vector<int> result = {first};
        for (const auto& x: encoded) {
            result.emplace_back(result.back() ^ x);
        }
        return result;
    }
};
